--- ncbiquery.py.orig	2020-12-08 23:10:43.000000000 +0100
+++ ncbiquery.py	2020-12-08 23:13:21.000000000 +0100
@@ -568,7 +568,7 @@
 
 def load_ncbi_tree_from_dump(tar):
     from ete2 import Tree
-    # Download: ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz
+    # Download: https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz
     parent2child = {}
     name2node = {}
     node2taxname = {}
@@ -641,7 +641,7 @@
     if not targz_file:
         import urllib
         print >>sys.stderr, 'Downloading taxdump.tar.gz from NCBI FTP site...'
-        urllib.urlretrieve("ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz", "taxdump.tar.gz")
+        urllib.urlretrieve("https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz", "taxdump.tar.gz")
         print >>sys.stderr, 'Done. Parsing...'
         targz_file = "taxdump.tar.gz"
 
@@ -687,7 +687,7 @@
         db.execute(cmd)
     print
 
-    db.execute("INSERT INTO stats (version) VALUES (%d);" %DB_VERSION)
+    db.execute("INSERT OR REPLACE INTO stats (version) VALUES (%d);" %DB_VERSION)
     db.commit()
     
     for i, line in enumerate(open("syn.tab")):
@@ -695,7 +695,7 @@
             print >>sys.stderr, '\rInserting synonyms:     % 6d' %i,
             sys.stderr.flush()
         taxid, spname = line.strip('\n').split('\t')
-        db.execute("INSERT INTO synonym (taxid, spname) VALUES (?, ?);", (taxid, spname))
+        db.execute("INSERT OR REPLACE INTO synonym (taxid, spname) VALUES (?, ?);", (taxid, spname))
     print
     db.commit()
     for i, line in enumerate(open("merged.tab")):
@@ -703,7 +703,7 @@
             print >>sys.stderr, '\rInserting taxid merges: % 6d' %i,
             sys.stderr.flush()
         taxid_old, taxid_new = line.strip('\n').split('\t')
-        db.execute("INSERT INTO merged (taxid_old, taxid_new) VALUES (?, ?);", (taxid_old, taxid_new))
+        db.execute("INSERT OR REPLACE INTO merged (taxid_old, taxid_new) VALUES (?, ?);", (taxid_old, taxid_new))
     print
     db.commit()
     for i, line in enumerate(open("taxa.tab")):
@@ -711,7 +711,7 @@
             print >>sys.stderr, '\rInserting taxids:      % 6d' %i,
             sys.stderr.flush()
         taxid, parentid, spname, common, rank, lineage = line.strip('\n').split('\t')
-        db.execute("INSERT INTO species (taxid, parent, spname, common, rank, track) VALUES (?, ?, ?, ?, ?, ?);", (taxid, parentid, spname, common, rank, lineage))
+        db.execute("INSERT OR REPLACE INTO species (taxid, parent, spname, common, rank, track) VALUES (?, ?, ?, ?, ?, ?);", (taxid, parentid, spname, common, rank, lineage))
     print
     db.commit()
 
